cbsurge/exposure/population/worldpop.py

```diff
@@ -261,21 +272,65 @@ async def get_links_from_table(data_id=None):
             return []
         return await extract_links_from_table(response.text)
 
-async def download(country_code=None, year=DATA_YEAR, force_reprocessing=False, download_path=None, sex=None, age_group=None):
+async def run_download(country_code=None, year=DATA_YEAR, download_path=None, sex=None, age_group=None, aggregate=True):
     """
     Args:
-        country_code: The country code of the country intended to be downloaded
-        year:
-        force_reprocessing:
-        download_path:
-        sex:
-        age_group:
 
+        country_code: The country code of the country intended to be downloaded
+        year: The year to download the data for
+        download_path: local download path
+        sex: The sex classification of the data to download
+        age_group: The age group classification of the data to download
+        aggregate: (bool) Whether to download the aggregate files
     Returns:
 
     """
-    # TODO: Should try to download data from azure, if the data is available
-    pass
+
+    async def __construct_file_name__():
+        if aggregate:
+            # construct the path for the aggregate files based of the WORLDPOP_AGE_MAPPING and get the label from the AGESEX_STRUCTURE_COMBINATIONS
+            for combo in AGESEX_STRUCTURE_COMBINATIONS:
+                if combo["age_group"] == age_group:
+                    return f"{AZ_ROOT_FILE_PATH}/{year}/{country_code}/aggregate", f"{country_code}_{combo['label']}.tif"
+        else:
+            # construct the path for the individual files checking where the age group falls in the WORLDPOP_AGE_MAPPING
+            pass
+    logging.info("Starting data download from azure")
+
+    session = Session()
+    credential = session.get_credential()
+
+    blob_service_client = BlobServiceClient(
```

Comment:
can you use `get_blob_service_client` method in session class instead?

Reply:
you also can use `get_blob_container_client` method to directory get blob container client from Session